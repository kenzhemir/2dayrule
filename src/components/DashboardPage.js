import React from "react";
import { connect } from "react-redux";
import { getVisibleRules } from "../redux/rules";
import { syncRules, unsyncRules, getSyncStatus } from "../redux/rulesSync";

const DashboardPage = props => (
	<>
		Your rules:
		<div>
			Listen to add is on: {props.syncStatus.toString()}
			<button
				onClick={() => {
					if (!props.syncStatus) {
						props.syncRules();
					} else {
						props.unsyncRules();
					}
				}}
			>
				turn sync {props.syncStatus ? "off" : "on"}
			</button>
		</div>
		<div>
			{props.rules.map(rule => (
				<div key={rule.id}>
					<h3>{rule.name}</h3>
					<p>{rule.description}</p>
				</div>
			))}
		</div>
	</>
);

const mapStateToProps = state => ({
	rules: getVisibleRules(state.rules),
	syncStatus: getSyncStatus(state)
});
const mapDispatchToProps = dispatch => ({
	syncRules: () => dispatch(syncRules()),
	unsyncRules: () => dispatch(unsyncRules())
});

export default connect(mapStateToProps, mapDispatchToProps)(DashboardPage);
