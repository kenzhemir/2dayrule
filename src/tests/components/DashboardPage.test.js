import { shallow } from "enzyme";
import React from "react";

import DashboardPage from "../../components/DashboardPage";

test("should render expense list with expenses", () => {
	const wrapper = shallow(<DashboardPage />);
	expect(wrapper).toMatchSnapshot();
});
