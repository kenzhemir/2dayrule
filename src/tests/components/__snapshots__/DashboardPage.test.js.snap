// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense list with expenses 1`] = `
<div>
  DashboardPage content
</div>
`;
